### ISSUE TITLE
[!!] Search should be under a heading & reorder the top portion

### ISSUE THREAD

**Issue by MicheleAWilliams-A11y** [w3c/wai-course-list#279]
The search field is sitting above a main heading for the courses (for which it's related) and below the "Important Disclaimer" heading (for which it is not related). This violates "Meaningful Sequence" and "Info and Relationships" and is easy to miss if navigating by heading with a screen reader. 

Additionally, the feedback for the search field is split up because you hear the dynamic heading (which I don't advise, see issue #278) but as you navigate (especially with a screen reader) you first hear the "Sort by" menu, then the heading (which sometimes doesn't announce as a heading when navigating with VoiceOver because it's also an alert), then you encounter the list of courses or the "...no courses match" text. This is really broken up and scattered - visually and with a screen reader. 

I advise a static heading of "Course List" followed by the "Sort by" menu then the "Search" field. You can also add a summary of how many choices are showing as plain text below these prior to listing the courses, or show the "no courses match" text. That way the next item in the order if you search are the results, and it's less confusing if you move from "Sort by" to "Search".

**Comment by leticiaseixas** [comment]
Hi @MicheleAWilliams-A11y, thanks for the feedback.

From what I understand there are two main issues here:

1. The first, about the "Important Disclaimer" can be better discussed in #285.

2. The second, as I understand it, you suggest adding: a main title to the results list, followed by the "Sort by", "Search" field, and a search result title/summary. Is that correct? If so, the search element consists of a filter element, like the others present in the sidebar. It seems strange to me to have a sort element in the middle of this.

**Comment by MicheleAWilliams-A11y** [comment]
Hi @leticiaseixas - my primary issue here was the placement of the Search field being above the Course List heading, not below it. It's usually advised that you place content below a heading.

Perhaps another issue to have others weigh in on.  cc: @daniel-montalvo and @Sylvie42 if you have time to comment

**Comment by leticiaseixas** [comment]
Right!
I suggested we discuss this further with the EOWG this Friday, it would also be interesting if you could better explain what your concerns are. 
Just to confirm and come up with a more solid idea of your proposal, it would then be something like these images below?

![Captura de ecrã 2022-07-27, às 09 23 15](https://user-images.githubusercontent.com/2460369/181199946-518654f4-57d7-4d69-a4bb-e94259fb8062.png)

![Captura de ecrã 2022-07-27, às 09 23 29](https://user-images.githubusercontent.com/2460369/181199955-4d2bd810-81e2-4ec1-bfff-3450aceae1ba.png)

**Comment by MicheleAWilliams-A11y** [comment]
Yes, that represents my proposal -- or even reusing your "Showing X of Y courses" heading and simply moving the search under that since folks said they didn't mind that as a heading (though I _really_ like what you have in these photos and I'll explain why below).

First, to explain why I flagged this as an issue: as I've always understood it, screen reader users in particular experience a new page by first navigating by headings to get a sense of the page. Then they start navigating _down_ from those headings. Thus on the current page (the one we reviewed), if a person navigates to the heading "Showing 3 of 5 courses" and moves down, they will miss the search field. (Of course there are other ways to find it but they're not as likely to happen as moving by headings.) Also in general, headings should describe the content below it; the search is related to the list of courses below it but appears _above_ the heading describing that content.

In addition, I believe this would also relate back to [Sylvie42](https://github.com/Sylvie42)'s point in ticket #278 where Slyvie said, 
> "But what is confusing is that there are 5 courses and you don't know how to see the two remaining courses. Do you have to modify filters?" 

I suspect using this design as you have it above (putting "Filters" and "Course List" as primary headings) will make the sections and their relationships even clearer for those who can't see the layout. Then the text "Showing X out of Y courses" isn't necessarily the main section heading but rather a subheading and/or supplemental information (and still an alert), but associated with the description of "Course list".

So those are my thoughts and reasons why I flagged it - hope this helps & makes sense!
